README.md

```diff
@@ -89,19 +94,96 @@ int main()
 
   try
   {
-    auto client = BlobClient::CreateFromConnectionString(
-        GetConnectionString(), containerName, blobName);
+    // There are many types of client types, use the one that's appropriate
+    // for your scenario, such as BlobContainerClient.
+    BlobClient client
+        = BlobClient::CreateFromConnectionString(GetConnectionString(), containerName, blobName);
+    BlockBlobClient block = client.AsBlockBlobClient();
+
+    // Assuming there is some Azure::Core::IO::BodyStream containing the data to upload.
+    // For example either a MemoryBodyStream or FileBodyStream.
+    std::vector<uint8_t> data;
+    Azure::Core::IO::MemoryBodyStream stream(data);
+
+    Azure::Response<Models::UploadBlockBlobResult> response = block.Upload(stream);
+
+    Models::UploadBlockBlobResult model = response.Value;
+    printf("Last modified date of uploaded blob: %s\n", model.LastModified.ToString().c_str());
   }
   catch (const Azure::Core::RequestFailedException& e)
   {
+    printf("Status Code: %d, Reason Phrase: %s\n", (int)e.StatusCode, e.ReasonPhrase.c_str());
     std::cout << e.what() << std::endl;
     return 1;
   }
   return 0;
 }
 ```
 
-Understanding the key concepts from the `azure-core-cpp` library, which is leveraged by all client libraries will also be helpful in getting started, regardless of which Azure service you want to use. You can find more information about them, with sample code snippets, here: https://github.com/Azure/azure-sdk-for-cpp/tree/master/sdk/core/azure-core#key-concepts
+#### Key Core concepts
+
+Understanding the key concepts from the `azure-core-cpp` library, which is leveraged by all client libraries will also be helpful in getting started, regardless of which Azure service you want to use.
+
+The main shared concepts of `Azure::Core` include:
+
+- Accessing HTTP response details for the returned model of any SDK client operation, via `Response<T>`.
+- Exceptions for reporting errors from service requests in a consistent fashion via the base exception type `RequestFailedException`.
+- Abstractions for Azure SDK credentials (`TokenCredential`).
+- Handling streaming data and input/output (I/O) via `BodyStream` along with its derived types.
+- Polling long-running operations (LROs), via `Operation<T>`.

```

Comment:
Add our paging support too?

Reply:
Yes! Good call out, we did finally add that.